src/sizzle.js

```diff
@@ -669,8 +669,9 @@ setDocument = Sizzle.setDocument = function( node ) {
 
 					// Fall back on getElementsByName
 					elems = context.getElementsByName( id );
+					length = elems.length;
 					i = 0;
-					while ( (elem = elems[i++]) ) {
+					for ( ; i !== length && (elem = elems[ i ]) != null; i++ ) {
```

Comment:
This is probably not an issue, but I feel like `i < length` is safer.

Reply:
`i < length` would definitely be an issue, because `length` might be a DOM node in IE<9.

Reply:
makes sense.

Reply:
Perhaps we could put in a comment relating that as a reminder.